src/sentry/snuba/discover.py

```diff
@@ -634,8 +644,24 @@ def top_events_timeseries(
             limit=10000,
             referrer=referrer,
         )
+        if len(top_events["data"]) == limit and len(result.get("data", [])) and include_other:
+            other_result = raw_query(
+                aggregations=snuba_filter.aggregations,
+                conditions=other_conditions,
```

Comment:
As discussed in slack, this condition is incorrect in some cases as we need to properly apply Demorgan's Law. But let's start with this for now and follow up with fixes.

Reply:
:+1: Added a TODO for this too, will be first thing I address when I tackle this next